### PR TITLE
Fixes #22098: Update deployment infrastructure code to apply metadata-complete attribute only to annotations with corresponding deployment descriptor element.

### LINKED ISSUE #22098: EE schema metadata-complete attribute should only be applicable to annotations without corresponding deployment descriptor element
EE schema metadata-complete attribute should only be applicable to annotations without corresponding deployment descriptor element.

### PR DESCRIPTION
**Pull request by honghzzhang**
Fixes #22098: Update deployment infrastructure code to apply metadata-complete attribute only to annotations with corresponding deployment descriptor element.

### CODE REVIEW

**Comment by honghzzhang** [comment]
@glassfishrobot Run CI tests please

File: appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/archivist/Archivist.java [hunk 1]
```diff
@@ -474,6 +474,14 @@ protected void readAnnotations(ReadableArchive archive, T descriptor,
                             RootDeploymentDescriptor o = extension.getKey().getDefaultDescriptor();
                             if( o != null ) {
                                 o.setModuleDescriptor(descriptor.getModuleDescriptor());
+                                // for the case of extension descriptor not 
+                                // present, set the metadata-complete attribute 
+                                // value of the extension descriptor using the 
+                                // metadata-complete value of main descriptor
+                                if (o instanceof BundleDescriptor) {
+                                    boolean isFullMain = descriptor.isFullAttribute();
```
Comment (saxenaks) [inline appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/archivist/Archivist.java]: Minor : Call it isFullAttrMain

**Review by saxenaks — COMMENTED** [review]
Changes look good to me!

**Review by vinayvishal — COMMENTED** [review]
Looks good!

**Comment by honghzzhang** [comment]
I have ran additional CTS RQ jobs for the following  (areas which I think might be impacted): 
appclient, connector, servlet, ejb30, ejb32, j2eetools, jaxws, webservices12, webservices13
and they all passed.

**Review by saxenaks — APPROVED** [review]

**Pull request merged**